racecar:
  internal:
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 50

    left_rear_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController
      joint: left_rear_wheel_joint
      pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
    right_rear_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController
      joint: right_rear_wheel_joint
      pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
      
    left_front_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController
      joint: left_front_wheel_joint
      pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}
    right_front_wheel_velocity_controller:
      type: effort_controllers/JointVelocityController
      joint: right_front_wheel_joint
      pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}


    # Position Controllers ---------------------------------------
    left_steering_position_controller:
      joint: left_front_wheel_turn_joint
      type: effort_controllers/JointPositionController
      pid: {p: 5.0, i: 0.0, d: 2}
    right_steering_position_controller:
      joint: right_front_wheel_turn_joint
      type: effort_controllers/JointPositionController
      pid: {p: 5.0, i: 0.0, d: 2}